demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,582 @@
+        };
+
+        auto age_net = cv::gapi::ie::Params<AgeGender> {
+            FLAGS_m_ag,                         // path to model
+            fileNameNoExt(FLAGS_m_ag) + ".bin", // path to weights
+            FLAGS_d_ag                          // device to use
+        }.cfgOutputLayers({ "age_conv3", "prob" });
+
+
+        auto hp_net = cv::gapi::ie::Params<HeadPose> {
+            FLAGS_m_hp,                         // path to model
+            fileNameNoExt(FLAGS_m_hp) + ".bin", // path to weights
+            FLAGS_d_hp                          // device to use
+        }.cfgOutputLayers({ "angle_y_fc", "angle_p_fc", "angle_r_fc" });
+
+        auto lm_net = cv::gapi::ie::Params<FacialLandmark> {
+            FLAGS_m_lm,                        // path to model
+            fileNameNoExt(FLAGS_m_lm) + ".bin",// path to weights
+            FLAGS_d_lm                         // device to use
+        }.cfgOutputLayers({ "align_fc3" });
+
+        auto emo_net = cv::gapi::ie::Params<Emotions> {
+            FLAGS_m_em,                         // path to model
+            fileNameNoExt(FLAGS_m_em) + ".bin", // path to weights
+            FLAGS_d_em                          // device to use
+        };
+
+        // Form a kernel package (including an OpenCV-based implementation of our
+        // post-processing) and a network package (holding our three networks).
+        auto kernels = cv::gapi::kernels<OCVPostProc>();
+        auto networks = cv::gapi::networks(det_net, age_net, hp_net, lm_net, emo_net);
+
+        cv::GStreamingCompiled stream = pipeline.compileStreaming(cv::compile_args(kernels, networks));
+
+        cv::Mat frame, ssd_res;
+        std::vector<cv::Rect> face_hub;
+        auto out_vector = cv::gout(frame, ssd_res, face_hub);
+
+        std::vector<cv::Mat> out_ages, out_genders;
+        if (!FLAGS_m_ag.empty()) out_vector += cv::gout(out_ages, out_genders);
+
+        std::vector<cv::Mat> out_y_fc, out_p_fc, out_r_fc;
+        if (!FLAGS_m_hp.empty()) out_vector += cv::gout(out_y_fc, out_p_fc, out_r_fc);
+
+        std::vector<cv::Mat> out_emotions;
+        if (!FLAGS_m_em.empty()) out_vector += cv::gout(out_emotions);
+
+        std::vector<cv::Mat> out_landmarks;
+        if (!FLAGS_m_lm.empty()) out_vector += cv::gout(out_landmarks);
+
+        Visualizer::Ptr visualizer;
+        if (!FLAGS_no_show || !FLAGS_o.empty()) {
+            visualizer = std::make_shared<Visualizer>(!FLAGS_m_ag.empty(), !FLAGS_m_em.empty(), !FLAGS_m_hp.empty(), !FLAGS_m_lm.empty());
+        } else {
+            std::cout<< "To close the application, press 'CTRL+C' here" << std::endl;
+        }
+
+        std::list<Face::Ptr> faces;
+        std::ostringstream out;
+        size_t framesCounter = 0;
+        size_t id = 0;
+        cv::VideoWriter videoWriter;
+
+        Timer timer;
+
+        slog::info << "Setting media source" << slog::endl;            
+        try {
+            if (FLAGS_i == "cam")
```

Comment:
The general convention is that a numeric value of `-i` means a camera with that ID, while any other value means a path. Please follow that.

Reply:
Done

Reply:
It seems like now you're just always treating it as a path? What about the camera support?

Reply:
Added `setInput()` function that support this logic
I hope its relevant solution

Reply:
It's a step in the right direction, but please remove the special treatment for `cam`.

Reply:
Removed